metals/src/main/scala/scala/meta/internal/metals/PostCompileCache.scala

```diff
@@ -0,0 +1,60 @@
+package scala.meta.internal.metals
+
+import java.util.Collections.singletonList
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicBoolean
+import ch.epfl.scala.{bsp4j => b}
+import scala.meta.internal.metals.MetalsEnrichments._
+
+final class PostCompileCache(buildServer: () => Option[BuildServerConnection]) {
+  private val initialized = new AtomicBoolean(false)
+  val mainClasses = new ConcurrentHashMap[String, b.ScalaMainClass]()
+
+  def isInitialized: Boolean = initialized.get()
+
+  def initialize(target: b.BuildTargetIdentifier): CompletableFuture[Void] =
+    initialize(singletonList(target))
+
+  def initialize(
+      targets: java.util.List[b.BuildTargetIdentifier]
+  ): CompletableFuture[Void] = {
+    clear()
+
+    buildServer() match {
+      case Some(connection) =>
+        val parameters = new b.ScalaMainClassesParams(targets)
+        val task = connection
+          .mainClasses(parameters)
+          .thenAccept(initializeMainClasses)
+
+        initialized.set(true)
+        task
+      case None =>
+        CompletableFuture.completedFuture(null)
```

Comment:
Can't we just do `CompletableFuture.completedFuture(())` instead of null?

Reply:
it was `Void`, so nope :(
but anyway, it is fixed now - changed to `Unit` ;)